kube-runtime/src/scheduler.rs

```diff
@@ -392,4 +400,56 @@ mod tests {
         scheduler.next().now_or_never().unwrap().unwrap();
         assert!(poll!(scheduler.next()).is_pending());
     }
+
+    #[tokio::test]
+    async fn scheduler_should_overwrite_message_with_soonest_version() {
+        // Message type that is always considered equal to itself
+        #[derive(Derivative, Eq, Clone, Debug)]
+        #[derivative(PartialEq, Hash)]
+        struct SingletonMessage(#[derivative(PartialEq = "ignore", Hash = "ignore")] u8);
+
+        pause();
+
+        let now = Instant::now();
+        let scheduler = scheduler(
+            stream::iter([
+                ScheduleRequest {
+                    message: SingletonMessage(1),
+                    run_at: now + Duration::from_secs(2),
+                },
+                ScheduleRequest {
+                    message: SingletonMessage(2),
+                    run_at: now + Duration::from_secs(1),
+                },
+            ])
+            .on_complete(sleep(Duration::from_secs(5))),
+        );
+        assert_eq!(scheduler.map(|msg| msg.0).collect::<Vec<_>>().await, vec![2]);
+    }
+
+    #[tokio::test]
+    async fn scheduler_should_not_overwrite_message_with_later_version() {
+        // Message type that is always considered equal to itself
+        #[derive(Derivative, Eq, Clone, Debug)]
+        #[derivative(PartialEq, Hash)]
+        struct SingletonMessage(#[derivative(PartialEq = "ignore", Hash = "ignore")] u8);
```

Comment:
could be pulled out in the the test mod

Reply:
Yeah, true.

Reply:
c60f8cce1bdf0f3a9ad68971eeeeb727af01762f